### PR TITLE
[Feature] use Array<DecimalV3> for DLA

### PR DESCRIPTION
**Pull request by zombee0**
before this pr, DLA will convert external Decimal to Decimal64 or Decimal128, but Decimal64 and Decimal128 are not compatible when they are used as the itemType of arrayType, this pr is used to open this feature.
we need a comprehensive test.

Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/ArrayType.java [hunk 1]
```diff
@@ -25,8 +25,12 @@ public ArrayType(Type itemType) {
         this.itemType = itemType;
     }
 
-    public ArrayType(Type itemType, boolean fromSubQuery) {
-        this.itemType = itemType;
+    public ArrayType(Type itemType, boolean fromDlaEnableDecimalV3) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/catalog/ArrayType.java:28]: Could you comment when `fromDlaEnableDecimalV3` is false?
Don't know the reason for this change. I see all the usage is passing true.

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/catalog/ArrayType.java:28]: > Could you comment when `fromDlaEnableDecimalV3` is false?
> Don't know the reason for this change. I see all the usage is passing true.

I set them all be true, and then we will  test them, if any one of them is not ok, we just let that one be false which means we don't support that one and give it an unknown type.

**Comment by zombee0** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 7 (71.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|2|3|66.67%|[250]|
|:large_blue_circle:|com/starrocks/catalog/ArrayType.java|3|4|75.00%|[30]|

File: fe/fe-core/src/main/java/com/starrocks/connector/ColumnTypeConverter.java [hunk 2]
```diff
@@ -342,7 +342,7 @@ public static int getVarcharLength(String typeStr) {
     }
 
     private static ArrayType fromHudiTypeToArrayType(Schema typeSchema) {
-        return new ArrayType(fromHudiType(typeSchema.getElementType()));
+        return new ArrayType(fromHudiType(typeSchema.getElementType()), true);
     }
 
     public static boolean validateHiveColumnType(Type type, Type otherType) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/ColumnTypeConverter.java:348]: lack of iceberg

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 8 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|2|3|66.67%|[250]|
|:large_blue_circle:|com/starrocks/catalog/ArrayType.java|3|4|75.00%|[30]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|1|1|100.00%|[]|

**Comment by dulong41** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 8 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|2|3|66.67%|[250]|
|:large_blue_circle:|com/starrocks/catalog/ArrayType.java|3|4|75.00%|[30]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|1|1|100.00%|[]|

**Review by DorianZheng — DISMISSED** [review]

**Comment by zombee0** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 5 / 6 (83.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/ArrayType.java|3|4|75.00%|[30]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|1|1|100.00%|[]|

**Comment by zombee0** [comment]
run starrocks_be_unittest

**Comment by zombee0** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 8 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|2|3|66.67%|[294]|
|:large_blue_circle:|com/starrocks/catalog/ArrayType.java|3|4|75.00%|[30]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 8 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|2|3|66.67%|[294]|
|:large_blue_circle:|com/starrocks/catalog/ArrayType.java|3|4|75.00%|[30]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|1|1|100.00%|[]|

**Comment by zombee0** [comment]
run starrocks_be_unittest

**Review by DorianZheng — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Youngwb — APPROVED** [review]

**Pull request merged**